tests/scene/test_node_3d.h

```diff
@@ -0,0 +1,307 @@
+
+	test_node3->set_position(Vector3(0, 0, 10));
+	test_node3->set_rotation(Vector3(0, 0, 0));
+	test_node3->set_scale(Vector3(2, 2, 2));
+
+	test_sibling->set_position(Vector3(5, 0, 10));
+	test_sibling->set_rotation(Vector3(0, 0, 0));
+	test_sibling->set_scale(Vector3(2, 2, 1));
+
+	SUBCASE("[Node3D] look_at") {
+		test_node3->look_at(Vector3(1, 0, 1));
+
+		CHECK(test_node3->get_global_position().is_equal_approx(Vector3(200, 0, 110)));
+		CHECK(test_node3->get_global_rotation().is_equal_approx(Vector3(0, 1.069691, 0)));
+		CHECK(test_node3->get_position().is_equal_approx(Vector3(0, 0, 10)));
+		CHECK(test_node3->get_rotation().is_equal_approx(Vector3(0, 1.069691, 0)));
+		CHECK(test_node3->get_scale().is_equal_approx(Vector3(2, 2, 2)));
+
+		test_node3->look_at(Vector3(0, 0, 10));
+
+		CHECK(test_node3->get_global_position().is_equal_approx(Vector3(200, 0, 110)));
+		CHECK(test_node3->get_global_rotation().is_equal_approx(Vector3(0, 1.107149, 0)));
+		CHECK(test_node3->get_position().is_equal_approx(Vector3(0, 0, 10)));
+		CHECK(test_node3->get_rotation().is_equal_approx(Vector3(0, 1.107149, 0)));
+		CHECK(test_node3->get_scale().is_equal_approx(Vector3(2, 2, 2)));
+
+		// Revert any rotation caused by look_at
+		test_node3->set_rotation(Vector3(0, 0, 0));
+
+		// look_at non-local target
+		Node3D *target_node = memnew(Node3D);
+		SceneTree::get_singleton()->get_root()->add_child(target_node);
+
+		target_node->set_position(Vector3(10, 10, 10));
+		test_node3->look_at(target_node->get_global_position());
+		CHECK(test_node3->get_global_rotation().is_equal_approx(Vector3(0.093183, 1.086318, 0)));
+
+		test_node3->set_rotation(Vector3(0, 0, 0));
+		memdelete(target_node);
+	}
+
+	SUBCASE("[Node3D] rotate_object_local") {
+		// Basic rotate_object_local tests
+		test_node3->rotate_object_local(Vector3(0, 1, 0), Math_PI);
+		CHECK(test_node3->get_global_position().is_equal_approx(Vector3(200, 0, 110)));
+		CHECK(wrap_rotation_angles(test_node3->get_global_rotation_degrees()).is_equal_approx(Vector3(0, 180, 0)));
+		CHECK(test_node3->get_position().is_equal_approx(Vector3(0, 0, 10)));
+		CHECK(wrap_rotation_angles(test_node3->get_rotation_degrees()).is_equal_approx(Vector3(0, 180, 0)));
+		CHECK(test_node3->get_scale().is_equal_approx(Vector3(2, 2, 2)));
+		CHECK_EQ(test_node3->get_rotation_edit_mode(), Node3D::ROTATION_EDIT_MODE_EULER);
+		CHECK_EQ(test_node3->get_rotation_order(), EulerOrder::YXZ);
+
+		// Change rotation order and rotate again
+		test_node3->set_rotation_order(EulerOrder::XYZ);
+		test_node3->rotate_object_local(Vector3(0, 0, -1), Math_PI / 2.0);
+		CHECK(test_node3->get_global_position().is_equal_approx(Vector3(200, 0, 110)));
+		CHECK(wrap_rotation_angles(test_node3->get_global_rotation_degrees()).is_equal_approx(Vector3(0, 180, 270)));
+		CHECK(test_node3->get_position().is_equal_approx(Vector3(0, 0, 10)));
+		CHECK(wrap_rotation_angles(test_node3->get_rotation_degrees()).is_equal_approx(Vector3(180, 0, 90)));
```

Comment:
Unfortunately, `posmod(360)` only works in the main cases and not in the corner case I have in this line of code. 
The corner case: When executing with floats, the value of the y axes is -0.000009. When executing with doubles, the value of the y axes is 0.00000000000001. That means in the case of posmod(360) for double, no modulo is applied and the value stays the same. In the case of float precision, since it has a negative value, the value is adapted to 359,999991, which will lead to a failing test result since the test checks for 0. I hope that makes the situation clear.
We can leave it like that, or I could remove this case and take other values to avoid this corner case. Then there is no need for the `wrap_rotation_angles()` method since it only appears in this specific case.

Reply:
@AThousandShips @akien-mga Should I just remove this case or change the values?